cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,217 @@
+"""
+Views for the maintenance app.
+"""
+import logging
+from django.db import transaction
+from django.core.urlresolvers import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import View
+
+from edxmako.shortcuts import render_to_response
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from contentstore.management.commands.utils import get_course_versions
+from util.views import require_global_staff
+
+
+log = logging.getLogger(__name__)
+
+MAINTENANCE_COMMANDS = {
+    "force_publish_course": {
+        "url": reverse_lazy("maintenance:force_publish_course"),
+        "name": _("Force Publish Course"),
+        "slug": "force_publish_course",
+        "description": _("Force publish course."),
+    },
+}
+
+
+COURSE_KEY_ERROR_MESSAGES = {
+    'empty_course_key': _('Please provide course id.'),
+    'invalid_course_key': _('Invalid course key.'),
+    'course_key_not_found': _('No matching course found.')
+}
+
+
+def get_maintenace_urls():
+    """
+    Returns all URLs for maintenance app.
+    """
+    url_list = []
+    for key, val in MAINTENANCE_COMMANDS.items():  # pylint: disable=unused-variable
+        url_list.append(val['url'])
+    return url_list
+
+
+class MaintenanceIndexView(View):
+    """
+    View for maintenance dashboard, used by the escalation team.
+    """
+
+    @method_decorator(require_global_staff)
+    def get(self, request):
+        """Render the maintenance index view. """
+        return render_to_response('maintenance/index.html', {
+            "commands": MAINTENANCE_COMMANDS,
+        })
+
+
+class MaintenanceBaseView(View):
+    """
+    Base class for Maintenance views.
+    """
+
+    template = 'maintenance/container.html'
+
+    def __init__(self, command=None):
+        self.context = {
```

Comment:
I am doing this because every view would have context variable with the exact same data initially. So it is better if we set it in BaseView's init method.